common/scala/src/main/scala/org/apache/openwhisk/core/database/elasticsearch/ElasticSearchActivationStore.scala

```diff
@@ -0,0 +1,482 @@
+              Right(DocInfo(bulkRes.id))
+            else
+              Left(PutException(
+                s"Unexpected error: ${bulkRes.error.map(e => s"${e.`type`}:${e.reason}").getOrElse("unknown")}, code: ${bulkRes.status} on 'bulk_put'"))
+          }
+        } else {
+          transid.failed(
+            this,
+            start,
+            s"'activations' failed to put documents, http status: '${res.status}'",
+            ErrorLevel)
+          throw new Exception("Unexpected http response code: " + res.status)
+        }
+      }
+
+    reportFailure(
+      res,
+      failure =>
+        transid
+          .failed(this, start, s"[PUT] 'activations' internal error, failure: '${failure.getMessage}'", ErrorLevel))
+  }
+
+  override def get(activationId: ActivationId, context: UserContext)(
+    implicit transid: TransactionId): Future[WhiskActivation] = {
+
+    cacheLookup(CacheKey(activationId.asString), doGet(activationId))
+  }
+
+  private def doGet(activationId: ActivationId)(implicit transid: TransactionId): Future[WhiskActivation] = {
+
+    val start =
+      transid.started(this, LoggingMarkers.DATABASE_GET, s"[GET] 'activations' finding activation: '$activationId'")
+
+    val index = generateIndex(activationId.toString.split("/")(0))
+    val res = client
+      .execute {
+        search(index) query { termQuery("_id", activationId.asString) }
+      }
+      .map { res =>
+        if (res.status == StatusCodes.OK.intValue) {
+          if (res.result.hits.total == 0) {
+            transid.finished(this, start, s"[GET] 'activations', document: '$activationId'; not found.")
+            throw NoDocumentException("not found on 'get'")
+          } else {
+            transid.finished(this, start, s"[GET] 'activations' completed: found activation '$activationId'")
+            restoreAnnotations(restoreResponse(res.result.hits.hits(0).sourceAsString.parseJson.asJsObject))
+              .convertTo[WhiskActivation]
+          }
+        } else if (res.status == StatusCodes.NotFound.intValue) {
+          transid.finished(this, start, s"[GET] 'activations', document: '$activationId'; not found.")
+          throw NoDocumentException("not found on 'get'")
+        } else {
+          transid
+            .finished(
+              this,
+              start,
+              s"[GET] 'activations' failed to get document: '$activationId'; http status: '${res.status}'")
+          throw new Exception("Unexpected http response code: " + res.status)
+        }
+      } recoverWith {
+      case _: DeserializationException => throw DocumentUnreadable(Messages.corruptedEntity)
+    }
+
+    reportFailure(
+      res,
+      failure =>
+        transid.failed(
+          this,
+          start,
+          s"[GET] 'activations' internal error, doc: '$activationId', failure: '${failure.getMessage}'",
+          ErrorLevel))
+  }
+
+  override def delete(activationId: ActivationId, context: UserContext)(
```

Comment:
Later we should review the usage for `delete` as activation records are immutable and removed via expiry instead of explicit removal. So technically we can drop the delete support